AB#2508 Create signed SBOMs for each build (#303)
edgelesssys/constellation: docs/docs/workflows/sbom.md

> # Consume Software Bill of Materials (SBOMs)

Reviewer: ```suggestion
# Consume software bill of materials (SBOMs)
```
I find both sentence-style and title-style capitalization equally often. As MS guide prefers sentence-style, I'd suggest this.

I'm also not sure whether the plural actually needs to be *bills*?
```suggestion
# Consume software bills of materials (SBOMs)
```

Author: I wen't with the first suggestion, double plural reads odd to me.

> Constellation produces [Software Bill of Materials (SBOMs)](https://www.ntia.gov/SBOM) for each generated artifact. See the [Components](../architecture/components.md) page for an overview and description of each.
> 

Reviewer: ```suggestion
Constellation builds produce a [software bill of materials (SBOM)](https://www.ntia.gov/SBOM) for each generated [artifact](../architecture/components.md).
```

> SBOMs can be used to make informed decisions about dependencies and vulnerabilities in a given application. Enterprises rely on SBOMs to maintain an inventory of used applications and allows them to take data-driven approaches to managing risks related to vulnerabilities.

Reviewer: ```suggestion
You can use SBOMs to make informed decisions about dependencies and vulnerabilities in a given application. Enterprises rely on SBOMs to maintain an inventory of used applications, which allows them to take data-driven approaches to managing risks related to vulnerabilities.
```

> All SBOMs are generated using [Syft](https://github.com/anchore/syft), signed using [Cosign](https://github.com/sigstore/cosign), and stored as close to the produced artifact as possible.

Reviewer: ```suggestion
SBOMs for Constellation are generated using [Syft](https://github.com/anchore/syft), signed using [Cosign](https://github.com/sigstore/cosign), and stored with the produced artifact.
```

Verification for ghcr.io/edgelesssys/constellation/verification-service@v2.2.0 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - The signatures were verified against the specified public key
$ jq -r .payload verification-service.sbom | base64 -d > verification-service.cyclonedx.sbom
```

:::note

This example considered only the `verification-service`. The same approach will work for all containers made available in the [Constellation container registry](https://github.com/orgs/edgelesssys/packages?repo_name=constellation).

:::

<!--
TODO: Once mkosi is implemented
## Operating System
-->

## Vulnerability scanning

A plethora of tools can be used to consume SBOMs. This section provides suggestions for tools that are popular and known to produce reliable results, but any tool that consumes [SPDX](https://spdx.dev/) or [CycloneDX](https://cyclonedx.org/) files should work!

Syft is able to [convert between the two formats](https://github.com/anchore/syft#format-conversion-experimental) in case you require a specific type.

### Grype

> [Grype] is a CLI based tool which lends itself very well for integration into CI/CD systems or local developer machines. It's also able to consume the signed attestation statement directly and does the verification in one go.

Reviewer 2: ```suggestion
[Grype] is a CLI based tool which lends itself well for integration into CI/CD systems or local developer machines. It's also able to consume the signed attestation statement directly and does the verification in one go.
```
